### PR TITLE
Merge 2.7.x into `main` before `2.8.0` release

### PR DESCRIPTION
**Pull request by dakshvar22**
**Proposed changes**:
- ...

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by dakshvar22** [comment]
@wochinge Can you please take a look at the changes to `poetry.lock` file? There was one merge conflict but that looked like new dependencies got added in `2.7.x`, so I brought them in as is.

File: poetry.lock [hunk 1]
```diff
@@ -4596,6 +4618,13 @@ python-engineio = [
     {file = "python-engineio-4.2.0.tar.gz", hash = "sha256:4e97c1189c23923858f5bb6dc47cfcd915005383c3c039ff01c89f2c00d62077"},
     {file = "python_engineio-4.2.0-py2.py3-none-any.whl", hash = "sha256:c6c119c2039fcb6f64d260211ca92c0c61b2b888a28678732a961f2aaebcc848"},
 ]
+<<<<<<< HEAD
+=======
```
Comment (dakshvar22) [inline poetry.lock:4622]:
```suggestion
```

File: poetry.lock [hunk 2]
```diff
@@ -4596,6 +4618,13 @@ python-engineio = [
...
+python-jose = [
+    {file = "python-jose-3.3.0.tar.gz", hash = "sha256:55779b5e6ad599c6336191246e95eb2293a9ddebd555f796a65f838f07e5d78a"},
+    {file = "python_jose-3.3.0-py2.py3-none-any.whl", hash = "sha256:9b1376b023f8b298536eedd47ae1089bcdb848f1535ab30555cd92002d78923a"},
+]
+>>>>>>> 2.7.x
```
Comment (dakshvar22) [inline poetry.lock:4627]:
```suggestion
```

**Comment by dakshvar22** [comment]
Okay def something up with the poetry lockfile (make installs have failed) but I am not sure what's wrong.

**Comment by dakshvar22** [comment]
Just saw this [section](https://github.com/RasaHQ/rasa#resolving-merge-conflicts). Will try it out now.

**Review by wochinge — APPROVED** [review]

**Comment by wochinge** [comment]
I'd just delete and run `poetry.lock` again

**Comment by dakshvar22** [comment]
@wochinge I deleted the lockfile, ran `poetry update` which created a new `poetry.lock` and pushed it. Packages still [don't seem to build](https://github.com/RasaHQ/rasa/pull/9072/checks?check_run_id=3017439204).

**Comment by dakshvar22** [comment]
Trying out a different [PR](https://github.com/RasaHQ/rasa/pull/9073) which used `poetry-merge-lock` to resolve conflicts in lockfile.

**Pull request merged**